test/conformance/chainsaw/validate/clusterpolicy/cornercases/variable-substitution-failure-messages/policy.yaml

```diff
@@ -0,0 +1,99 @@
+spec:
+  validationFailureAction: enforce
+  background: true
+  rules:
+    - name: check-runasuser
+      context:
+      - name: allowedUIDs
+        variable:
+          value: "9999 | 4000"
+      match:
+        any:
+        - resources:
+            kinds:
+              - Pod
+      validate:
+        message: >-
+          Running with specific user IDs {{ allowedUIDs }}. The fields
+          spec.securityContext.runAsGroup, spec.containers[*].securityContext.runAsGroup,
+          spec.initContainers[*].securityContext.runAsGroup, and
+          spec.ephemeralContainers[*].securityContext.runAsGroup must be
+          set to one of the {{ allowedUIDs }} values.
+        anyPattern:
+        - spec:
+            securityContext:
+              runAsUser: "{{ allowedUIDs }}"
+            =(ephemeralContainers):
+              - =(securityContext):
+                  =(runAsUser): "{{ allowedUIDs }}"
+            =(initContainers):
+              - =(securityContext):
+                  =(runAsUser):  "{{ allowedUIDs }}"
+            containers:
+              - =(securityContext):
+                  =(runAsUser): "{{ allowedUIDs }}"
+        - spec:
+            =(ephemeralContainers):
+              - securityContext:
+                  runAsUser: "{{ allowedUIDs }}"
+            =(initContainers):
+              - securityContext:
+                  runAsUser: "{{ allowedUIDs }}"
+            containers:
+              - securityContext:
+                  runAsUser: "{{ allowedUIDs }}"
+    - name: check-runasgroup
+      context:
+      - name: allowedGIDs
+        variable:
+          value: "3000 | 3004"
+      match:
+        any:
+        - resources:
+            kinds:
+              - Pod
+      validate:
+        message: >-
+          Running with specific group IDs {{ allowedGIDs }} is allowed. The fields
+          spec.securityContext.runAsGroup, spec.containers[*].securityContext.runAsGroup,
+          spec.initContainers[*].securityContext.runAsGroup, and
+          spec.ephemeralContainers[*].securityContext.runAsGroup must be
+          set to one of the {{ allowedGIDs }} values.
+        anyPattern:
+        - spec:
+            securityContext:
+             runAsGroup: "{{ allowedGIDs }}"
+            =(ephemeralContainers):
+              - =(securityContext):
+                  =(runAsGroup): "{{ allowedGIDs }}"
+            =(initContainers):
+              - =(securityContext):
+                  =(runAsGroup): "{{ allowedGIDs }}"
+            containers:
+              - =(securityContext):
+                  =(runAsGroup): "{{ allowedGIDs }}"
+        - spec:
+            =(ephemeralContainers):
+              - securityContext:
+                  runAsGroup: "{{ allowedGIDs }}"
+            =(initContainers):
+              - securityContext:
+                  runAsGroup: "{{ allowedGIDs }}"
+            containers:
+              - securityContext:
+                  runAsGroup: "{{ allowedGIDs }}"
```

Comment:
The provided resource uses the user ID so there is no need for a rule that checks the group IDs. You can remove it.